cmd/restic/cmd_dump.go

```diff
@@ -74,56 +74,138 @@ func splitPath(p string) []string {
 	return append(s, f)
 }
 
-func printFromTree(ctx context.Context, tree *restic.Tree, repo restic.BlobLoader, prefix string, pathComponents []string, d *dump.Dumper, canWriteArchiveFunc func() error) error {
+func preparePathList(pathComponentsList []string) [][]string {
```

Comment:
Please add a short description how this function works. It took me a while to reverse engineer the underlying idea: first all paths are stored in a pathSet in sanitized form. The filtered set then only contains paths, for which no prefix already exists in the pathSet. This ensures that the order in which paths are processed does not matter.